src/exportJarSteps/ExportJarTaskProvider.ts

```diff
@@ -29,14 +46,74 @@ export class ExportJarTaskProvider implements TaskProvider {
         return task;
     }
 
+    private static exportJarTaskProvider: Disposable | undefined;
+
+    private tasks: Task[] | undefined;
+
     public async resolveTask(_task: Task): Promise<Task> {
-        return _task;
+        const definition: IExportJarTaskDefinition = <IExportJarTaskDefinition>_task.definition;
+        const folder: WorkspaceFolder = <WorkspaceFolder>_task.scope;
+        const stepMetadata: IStepMetadata = {
+            entry: undefined,
+            workspaceFolder: folder,
+            steps: [],
+        };
+        const task: Task = new Task(definition, folder, _task.name, ExportJarTaskProvider.exportJarType,
```

Comment:
Will `definition`/`folder` be undefined?

Reply:
Neither. `definition` and `folder` come from the pre-defined task generated by `provideTasks()` both.